tests/many_to_one/tests.py

```diff
@@ -666,3 +666,9 @@ def test_cached_relation_invalidated_on_save(self):
         self.a.reporter_id = self.r2.pk
         self.a.save()
         self.assertEqual(self.a.reporter, self.r2)
+
+    def test_cached_relation_with_to_field_not_cleared(self):
+        parent = Parent.objects.create(name='a')
+        child = ToFieldChild.objects.create(parent=parent)
+        with self.assertNumQueries(0):
+            self.assertEqual(child.parent.name, 'a')
```

Comment:
I guess you could use `self.assertIs(child.parent, parent)` to make sure the object was not recreated as well.